crates/biome_css_parser/src/syntax/mod.rs

```diff
@@ -79,13 +80,192 @@ pub(crate) fn parse_rule_block(p: &mut CssParser) -> ParsedSyntax {
     }
     let m = p.start();
     p.expect(T!['{']);
-    let list = p.start();
-    list.complete(p, CSS_DECLARATION_LIST);
+    parse_declaration_list(p);
     p.expect(T!['}']);
 
     Present(m.complete(p, CSS_BLOCK))
 }
 
+pub(crate) fn parse_declaration_list(p: &mut CssParser) {
+    let m = p.start();
+
+    // first parse declaration item
+    parse_declaration_item(p);
+    loop {
+        if !p.eat(T![;]) {
+            break;
+        }
+        parse_declaration_item(p);
+    }
+    m.complete(p, CSS_DECLARATION_LIST);
+}
+
+// CssDeclaration =
+// 	name: CssIdentifier | CssCustomProperty
+// 	':'
+// 	value: AnyCssValue
+// 	important: CssDeclarationImportant?
+#[inline]
+pub(crate) fn parse_declaration_item(p: &mut CssParser) {
+    if !is_at_identifier(p) {
+        return;
+    }
+    let m = p.start();
+    // name
+    parse_regular_identifier(p).ok();
+
+    p.expect(T![:]);
+    loop {
+        let any_css_value = parse_any_css_value(p);
+        if any_css_value.is_absent() {
+            break;
+        }
+    }
+    parse_declaration_important(p);
+    m.complete(p, CSS_DECLARATION);
+}
+
+#[inline]
+pub(crate) fn parse_declaration_important(p: &mut CssParser) {
+    if p.eat(T![!]) {
+        p.expect(T![important]);
+    }
+}
+#[inline]
+pub(crate) fn is_any_css_value(p: &mut CssParser) -> bool {
+    is_at_css_any_function(p)
+        || is_at_identifier(p)
+        || p.at(CSS_STRING_LITERAL)
+        || is_at_css_dimension(p)
+        || p.at(CSS_NUMBER_LITERAL)
+        || is_at_css_custom_property(p)
+        || is_at_css_ratio(p)
+}
+
+#[inline]
+pub(crate) fn parse_any_css_value(p: &mut CssParser) -> ParsedSyntax {
+    let css_any_function = parse_css_any_function(p);
```

Comment:
I'd like to propose the following structure:
```
if is_at_ration(p) {
  parse_ration(p)
} else if is_at_custom_property(p) {
  parse_custom_property(p)
}
```

Reply:
ok